shell/common/shell_test.h

```diff
@@ -31,22 +31,29 @@ namespace testing {
 
 class ShellTest : public FixtureTest {
  public:
+  struct Config {
+    // Required.
+    const Settings& settings;
+    // Defaults to GetTaskRunnersForFixture().
+    std::optional<TaskRunners> task_runners = {};
```

Comment:
This is where I think the builder could be helpful - you don't want the public interface here to actually be optional.

But if you don't want to use a builder I think we should make the `std::optional` be on the constructor for this struct and make the member non-optional - because otherwise callers will always have to check (and get linted against not checking now) whether the optional has a value when it really always does.

Reply:
I can create a `TaskRunnersBuilder`, which has two ctors:
```
TaskRunnersBuilder(TaskRunner); // implicit conversion
TaskRunnersBuilder(); // Defaults to GetTaskRunnersForFixture()
```

But I think it's really just reinventing the wheel for `std::optional` because it's not semantically obvious what the default behavior is (and it's now hidden even deeper).

And what's worse, because this field is a value, not a pointer of a function, I can't really specify an "empty" value if not for the default ctor.

> because otherwise callers will always have to check (and get linted against not checking now) whether the optional has a value when it really always does

I don't think this is a problem, because this struct really is only used by `ShellTest::CreateShell`.

Reply:
Another option is to mark `task_runners` as mandatory so that you always explicit write `GetTaskRunnersForFixture()` for every call.